### ISSUE TITLE
V4 broken for TS `nodenext` users

### ISSUE THREAD

**Issue by trevor-scheer** [apollographql/apollo-server#6730]
Lots of errors like this:
```
node_modules/@apollo/server/dist/esm/externalTypes/plugins.d.ts:6:344 - error TS2835: Relative import paths need explicit file extensions in EcmaScript imports when '--moduleResolution' is 'node16' or 'nodenext'. Did you mean './requestPipeline.js'?
```

Repro: https://codesandbox.io/s/musing-albattani-s8v12i

Looks like we just need extensions _everywhere_, not just runtime imports.

**Comment by trevor-scheer** [comment]
Closed via #6731

**Comment by jaydenseric** [comment]
The same issue desperately needs to be fixed in `@apollo/client`: https://github.com/apollographql/apollo-client/issues/9890 .
